_posts/2022-02-23-#24098.md

```diff
@@ -10,14 +10,30 @@ status: upcoming
 commit:
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+**This PR is a follow-up from [#17631](https://bitcoincore.reviews/17631), feel free to have a look at those notes first.**
+
+In RESTful APIs, [typically](https://rapidapi.com/blog/api-glossary/parameters/query/) path parameters  (e.g. `/some/unique/resource/`) are used to represent resources, and query parameters (e.g. `?sort=asc`) are used to control how these resources are being loaded through e.g. sorting, pagination, filtering, ...
+
+As first [discussed in #17631](https://github.com/bitcoin/bitcoin/pull/17631#discussion_r733031180), the [current REST api](https://github.com/bitcoin/bitcoin/blob/master/doc/REST-interface.md) contains two endpoints `/headers/` and `/blockfilterheaders/` that rather unexpectedly use path parameters to control how many (filter) headers are returned in the response. While this is no critical issue, it is unintuitive and we are still early enough to easily phase this behaviour out and ensure new endpoints (if any) do not have to stick to non-standard behaviour just for internal consistency.
+
+In this PR, a new `HTTPRequest::GetQueryParameter` method is introduced to easily parse query parameters, as well as two new `/headers/` and `/blockfilterheaders/` endpoints that use a count query parameter are introduced. The old path parameter-based endpoints are kept without too much overhead, but the documentation now points to the new query parameter-based endpoints as the default interface to encourage standardness.
+
 ## Questions
 1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?
--->
+1. In HTTP requests, what is the difference between a `path` and a `query` parameter?
+1. What are the **benefits** of changing the `COUNT` parameter from a `path` parameter to a `query` parameter?
+1. What are the **drawbacks** of changing the `COUNT` parameter from a `path` parameter to a `query` parameter?
+1. *(general discussion)* If the effort of implementing a change is already done, what could be some good reasons for rejecting an (unharmful) code change anyway? Do you (dis)agree with [this comment](https://github.com/bitcoin/bitcoin/pull/24098#pullrequestreview-860377099)?
+1. Does this PR change any of the existing function signatures? If so, why? Can this cause any behaviour change?
+1. Can you list all the commits that introduce behaviour change(s)? Do you feel comfortable about these behaviour change(s)?
+1. Consider the request (signet) `GET /rest/blockfilterheaders/basic/2/0000004c6aad0c89c1c060e8e116dcd849e0554935cd78ff9c6a398abeac6eda.json?count=1`. What would the response be prior to this PR? What would the response be after this PR? Which (modified) function is responsible for this behaviour change? Try reasoning about it before verifying experimentally.
```

Comment:
I recommend giving instructions to attendees about how to test locally, I imagine most people haven't tried using this interface.

Reply:
Added:

### Using the REST API

We'll test the [REST API](https://github.com/bitcoin-core-review-club/bitcoin/blob/pr24098/doc/REST-interface.md) on signet. To do so, first start a bitcoind instance with the `-rest` flag enabled. To use the `blockfilterheaders` endpoint, you'll also need to set `-blockfilterindex=1`:
`./bitcoind -signet -rest -blockfilterindex=1`

As soon as bitcoind is fully up and running, you should be able to query the API, for example by using curl on the command line:
`curl "127.0.0.1:38332/rest/chaininfo.json"`
As a response, you should get:
```
{"chain":"signet","blocks":78071,"headers":78071,"bestblockhash":"000000b3e98c0de440154f42819b56586ed36bad0baa2db8ba5d0950e416dcad","difficulty":0.002873067874458486,"time":1645188967,"mediantime":1645188099,"verificationprogress":0.9999934105995945,"initialblockdownload":false,"chainwork":"000000000000000000000000000000000000000000000000000000dc187f6fa0","size_on_disk":373918704,"pruned":false,"softforks":{"bip34":{"type":"buried","active":true,"height":1},"bip66":{"type":"buried","active":true,"height":1},"bip65":{"type":"buried","active":true,"height":1},"csv":{"type":"buried","active":true,"height":1},"segwit":{"type":"buried","active":true,"height":1},"taproot":{"type":"bip9","bip9":{"status":"active","start_time":-1,"timeout":9223372036854775807,"since":0,"min_activation_height":0},"height":0,"active":true}},"warnings":"This is a pre-release test build - use at your own risk - do not use for mining or merchant applications"}
```